attestation-tool/src/ecdsa.rs

```diff
@@ -0,0 +1,369 @@
+aW9uMRQwEgYDVQQHDAtTYW50YSBDbGFyYTELMAkGA1UECAwCQ0ExCzAJBgNVBAYT
+AlVTMFkwEwYHKoZIzj0CAQYIKoZIzj0DAQcDQgAEC6nEwMDIYZOj/iPWsCzaEKi7
+1OiOSLRFhWGjbnBVJfVnkY4u3IjkDYYL0MxO4mqsyYjlBalTVYxFP2sJBK5zlKOB
+uzCBuDAfBgNVHSMEGDAWgBQiZQzWWp00ifODtJVSv1AbOScGrDBSBgNVHR8ESzBJ
+MEegRaBDhkFodHRwczovL2NlcnRpZmljYXRlcy50cnVzdGVkc2VydmljZXMuaW50
+ZWwuY29tL0ludGVsU0dYUm9vdENBLmRlcjAdBgNVHQ4EFgQUImUM1lqdNInzg7SV
+Ur9QGzknBqwwDgYDVR0PAQH/BAQDAgEGMBIGA1UdEwEB/wQIMAYBAf8CAQEwCgYI
+KoZIzj0EAwIDSQAwRgIhAOW/5QkR+S9CiSDcNoowLuPRLsWGf/Yi7GSX94BgwTwg
+AiEA4J0lrHoMs+Xo5o/sX6O9QWxHRAvZUGOdRQ7cvqRXaqI=
+-----END CERTIFICATE-----"#;
+lazy_static::lazy_static! {
+    static ref PCS_TRUST_ROOT: MbedtlsList<Certificate> = {
+        let mut cert_chain = MbedtlsList::new();
+        let raw_cert = CString::new(PCS_TRUST_ROOT_CERT.as_bytes()).unwrap();
+        let cert = Certificate::from_pem(raw_cert.as_bytes_with_nul()).unwrap();
+        cert_chain.push(cert);
+
+        cert_chain
+    };
+}
+
+// OIDs for PCK X509 certificate extensions.
+const PCK_SGX_EXTENSIONS_OID: &[u64] = &[1, 2, 840, 113741, 1, 13, 1];
+const PCK_SGX_EXTENSIONS_FMSPC_OID: &[u64] = &[1, 2, 840, 113741, 1, 13, 1, 4];
+const PCK_SGX_EXTENSIONS_TCB_OID: &[u64] = &[1, 2, 840, 113741, 1, 13, 1, 2];
+
+// TCB url. // TODO: configurable.
+const TCB_URL: &str = "https://api.trustedservices.intel.com/sgx/certification/v4/tcb";
+const QE_IDENTITY_URL: &str =
+    "https://api.trustedservices.intel.com/sgx/certification/v4/qe/identity";
+
+#[derive(Clone, Debug, Default, serde::Deserialize)]
+/// Response from TCBInfo API.
+struct TCBInfoResponse {
+    #[serde(rename = "tcbInfo")]
+    pub tcb_info: TCBInfo,
+}
+
+// Taken from TCBLevel.matches (which is a private function).
+pub fn tcb_matches(tcb_level: &TCBLevel, tcb_comp_svn: &[u32], pcesvn: u32) -> bool {
+    // a) Compare all of the SGX TCB Comp SVNs retrieved from the SGX PCK Certificate (from 01 to
+    //    16) with the corresponding values in the TCB Level. If all SGX TCB Comp SVNs in the
+    //    certificate are greater or equal to the corresponding values in TCB Level, go to b,
+    //    otherwise move to the next item on TCB Levels list.
+    for (i, comp) in tcb_level.tcb.sgx_components.iter().enumerate() {
+        // At least one SVN is lower, no match.
+        if tcb_comp_svn[i] < comp.svn {
+            return false;
+        }
+    }
+
+    // b) Compare PCESVN value retrieved from the SGX PCK certificate with the corresponding value
+    //    in the TCB Level. If it is greater or equal to the value in TCB Level, read status
+    //    assigned to this TCB level. Otherwise, move to the next item on TCB Levels list.
+    if tcb_level.tcb.pcesvn < pcesvn {
+        return false;
+    }
+
+    // Match.
+    true
+}
+
+#[derive(Clone, Debug, Default, serde::Deserialize)]
+struct QEIdentityResponse {
+    #[serde(rename = "enclaveIdentity")]
+    pub enclave_identity: QEIdentity,
+}
+
+// Taken from QEIdentity.verify.
```

Comment:
Let's update Oasis Core to expose this so we can reuse it here in a later release.